### PR TITLE
Admit tree reroot transport projection before execution

### LINKED ISSUE #2583: Anti-pattern: Accepted requests can fail after execution because transport fit is not admitted
## Summary

Admission validated the *mathematical* request and the kernel returned the exact result — but the result then failed **transport composition**: JSON nesting depth, 2^53 interoperable-integer range, output-byte echo, or a Python host error inside result construction. An accepted `math.run` request turned into a host exception, which AGENTS.md forbids ("Every accepted request must return a typed result rather than expose a backend or host exception").

Distinct from #2575 (undercounted work): here the budget is right; the *transport envelope* is simply not part of admission.

## Concrete instances (all fixed on their PRs, pattern remains)

- Symbolic product admitted at exactly 512 work units produced a canonical result of >512 terms, violating `SymbolicMatrix`'s own validator after acceptance (#2401); sibling: scalar fast path undercharged retained denominators so real results exceeded the aggregate budget (#2401).
- Unifying two shallow terms composes an MGU binding that exceeds the JSON nesting-depth limit during `OperationResult` canonicalization (#2413).
- Quadratic-form evaluation admitted |x|,|y| up to 13.5e9 with coefficients to 10^6; exact values beyond 2^53-1 failed `canonicalize_json` (#2437).
- Independence-number requests passed input admission but their echoed witness/graph serialization exceeded `max_output_bytes` at dispatch (#2404).
- Dimension-zero encoders (real producer output of shorten) hit an IndexError in dual/parity-check kernels (#2324); oversized cross-sums raised UnboundLocalError inside validation (#2364).

## Why it keeps happening

Transport constraints live in dispatch/canonicalization layers (`CanonicalLimits`, `OperationResult.require_canonical_output`) while admission lives in domain models; no single test composes them.

## Remediation direction

- Add a shared property test harness: for every operation, boundary-admitted fixtures must survive serialize → parse → `require_canonical_output` → dispatch round-trip. "Accepted ⇒ transportable" becomes an enforced invariant instead of a review discovery.
- Where producer outputs can legitimately exceed transport limits, make it **admission's** job to bound them (result-sensitive envelopes like #2437's exact-value bound), not dispatch's failure.
- Adversarial test class: minimal accepted inputs engineered so each transport dimension (depth, int53, bytes, echo duplication) lands exactly one past its limit.

### PR DESCRIPTION
**Pull request by morluto**
Fixes #2583.

## Problem

A valid 256-node reroot request with long labels fit the request envelope but produced a 13,843,258-byte canonical result. Dispatch ran the kernel and then raised an `OperationResult` validation error because transport fit had not been admitted.

## Solution

Calculate the exact strict-JSON reroot result size with a linear tree traversal during `RerootRequest` admission. Oversized results now fail before execution with `graph.reroot_result_exceeds_transport_limit`, without materializing repeated reroot paths.

## Regression coverage

The catalog-driven regression covers serialize → parse → canonical projection → dispatch for an admitted request, and proves the oversized request returns the typed owner error rather than executing and failing during result validation. The size calculation was cross-checked against 32 varied tree shapes.

## Testing

- graph owner suite — 549 passed
- catalog and advertised examples — 654 passed
- tree-decomposition dispatch regression — 2 passed
- `make lint` and focused mypy — passed
- `git diff --check origin/main...HEAD` — passed

## Public contract impact

Oversized reroot results are now rejected at request admission with a typed error instead of exposing a post-execution validation failure.

## Checklist

- [ ] `make check` passes (not run)

### CODE REVIEW

File: src/jacobian/math/graphs/tree_decompositions/_models.py [hunk 1]
```diff
@@ -2,15 +2,100 @@
...
+        current = queue.popleft()
+        for neighbor in adjacency[current]:
+            if neighbor == parent[current] or neighbor == root_index:
+                continue
+            parent[neighbor] = current
+            depth[neighbor] = depth[current] + 1
+            traversal.append(neighbor)
+            queue.append(neighbor)
+
+    encoded_nodes = [len(encode_strict_json(node)) for node in decomposition.tree_nodes]
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/graphs/tree_decompositions/_models.py]: Root cause confirmed: `encode_strict_json` measures raw spellings while `canonicalize_json` NFC-normalizes before measuring, so decomposed combining marks inflated the admission estimate past the transport cap. Fixed in 319607241: `_reroot_result_wire_bytes` normalizes every label occurrence with the same NFC form the delivery boundary uses, so the estimate can no longer reject a payload whose canonical projection fits. Regression tests admit a 256-node decomposed-label path that previously failed measurement and still reject a genuinely oversized canonical projection.

File: src/jacobian/math/graphs/tree_decompositions/_models.py [hunk 2]
```diff
@@ -2,15 +2,105 @@
...
+            if neighbor == parent[current] or neighbor == root_index:
+                continue
+            parent[neighbor] = current
+            depth[neighbor] = depth[current] + 1
+            traversal.append(neighbor)
+            queue.append(neighbor)
+
+    normalized_nodes = [
+        unicodedata.normalize("NFC", node) for node in decomposition.tree_nodes
+    ]
```
[the remark this answers is not in the document]
Comment (morluto) [reply src/jacobian/math/graphs/tree_decompositions/_models.py]: Root cause confirmed: after switching measurement to NFC forms, two distinct raw spellings that normalize to the same string would pass admission yet collide as result-map keys at `canonicalize_json`, surfacing as a delivery-time validation exception instead of a typed request error. Fixed in dff3f4268: `RerootRequest` admission now compares labels after the transport boundary's NFC normalization and rejects collisions with `graph.reroot_tree_node_labels_collide_after_normalization` before execution. Regression test covers a precomposed/decomposed pair that `TreeDecomposition` accepts but reroot must reject.
